src/index.ts

```diff
@@ -172,10 +172,57 @@ app.get("/me", async (c) => {
     login.account.id,
   );
 
-  return c.json({
-    name,
-    email: login.account.email,
-  });
+  //roleを確認
+  const roleEntry = await c.env.DB
+  .prepare("SELECT role FROM account WHERE id = ?")
+  .bind(login.account.id)
+  .first();
+  if ( roleEntry === null ){
+    throw new Error("Entry was invalid");
+  }
+
+  const { role } = roleEntry;
+  //学生であった場合registrationから科目idを取り出す
+  if ( role === 'STUDENT'){
+    const subjectEntry = c.env.DB
+    .prepare("SELECT subject_id FROM registration WHERE sutudent_id = ?")
+    .bind(login.account.id)
+    .raw();
+
+    if ( subjectEntry === null ){
+      throw new Error("Entry was invalid");
+    }
+
+    const subject = subjectEntry;
+    
+    return c.json({
+      name,
+      email: login.account.email,
+      registrations: subject,
+    });
```

Comment:
```suggestion
    const subjectIdRows = await c.env.DB
      .prepare("SELECT subject_id FROM registration WHERE sutudent_id = ?")
      .bind(login.account.id)
      .raw();
    
    return c.json({
      name,
      email: login.account.email,
      registrations: subjectIdRows.flat(),
    });
```
[`raw` メソッド](https://developers.cloudflare.com/d1/platform/client-api/#await-stmtraw) はクエリ結果の値を順に詰めた配列を返すので, 仕様で指定したレスポンスにはなりません. この `subjectEntry` は `[ [ "xxxx" ], [ "yyyy" ] ]` のような配列になってしまいます.
そこで, [`flat` メソッド](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) を使ってネストを 1 段階減らしてやれば, 返すべき ID の配列になるでしょう.
`TEACHER` の方も同様にお願いします.